cypress-io/cypress
# fix: login/utm param bug fixes and test coverage

- Closes #23779 and 
- Reflects issues discovered during #23762 - I've extracted the new testing from that PR so that we can merge this on its own.
- Does some of the work needed for #23769 but does not exhaustively check UTM params (simply for reasons of time).

### User facing changelog


Users can now log in from the "Choose a browser" page of the Launchpad even if a browser is not open.

### Additional details


This PR combines a few related changes for login and UTM parameters. It addresses 4 things:

1. Can't log in from the "Choose a browser" page of the launchpad 
2. Hardcoded UTM parameters in some components that need to be contextual instead
3. Some testing TODOs that I hit when looking for test coverage related to UTM params, where we just were not testing them yet.
4. Some inaccuracies in tests themselves, for example testing the _header_ login button on the "logged out" state of the Runs page, not the in-page login button that's supposed to be there

### Steps to test


- In the Launchpad, advance to the "Choose a browser" page but do not open a browser. Use the Log In button in the header to log in. It should complete the login process and refocus the launchpad.
- For the UTM params, these are tedious to test manually, but in many cases can be seen with DEBUG logs turned on, so spot checking logging in and creating a project from various pages would be good.

### How has the user experience changed?


### PR Tasks
